CMakeLists.txt

```diff
@@ -0,0 +1,252 @@
+# SPDX-License-Identifier: Unlicense
+#
+# LibTomMath, a free open source portable number theoretic multiple-precision
+# integer (MPI) library written entirely in C.
+#
+
+cmake_minimum_required(VERSION 3.10)
+
+project(tommath
+    VERSION 1.2.0
+    DESCRIPTION "A free open source portable number theoretic multiple-precision integer (MPI) library written entirely in C."
+    HOMEPAGE_URL "https://www.libtom.net/LibTomMath"
+    LANGUAGES C)
+
+# package release version
+# bump if re-releasing the same VERSION + patches
+# set to 1 if releasing a new VERSION
+set(PACKAGE_RELEASE_VERSION 1)
+
+#-----------------------------------------------------------------------------
+# Include cmake modules
+#-----------------------------------------------------------------------------
+include(GNUInstallDirs)
+include(CheckIPOSupported)
+include(CMakePackageConfigHelpers)
+# default is "No tests"
+option(BUILD_TESTING "" OFF)
+include(CTest)
+include(sources.cmake)
+
+# The only direct cmake argument for now
+option(BUILD_SHARED_LIBS "Build shared library and only the shared library if \"ON\", default is static" OFF)
+
+#-----------------------------------------------------------------------------
+# Compose CFLAGS
+#-----------------------------------------------------------------------------
+
+# Some information ported from makefile_include.mk
+
+
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE "Release")
```

Comment:
```suggestion
  set(CMAKE_BUILD_TYPE "Release" CACHE STRING "CMake build type")
```

Reply:
IIUC all the currently failing builds are caused by this `CACHE STRING "CMake build type"` option ...

Reply:
Weird.
Let's focus on this github action run: https://github.com/libtom/libtommath/actions/runs/2024483993

All errors are due to missing symbols.

Reply:
Seems like you fixed it somehow?
https://github.com/libtom/libtommath/actions/runs/2024551731

Reply:
e.g. https://github.com/libtom/libtommath/runs/5650110146?check_suite_focus=true

clang requires at least `-O1` to build library and tests, gcc only requires at least `-O1` for the tests

As soon as I add the `CACHE STRING "CMake build type"` the default `CMAKE_BUILD_TYPE` isn't taken into account, but still stays at `""`. I've just removed it again and that build now succeeded https://github.com/libtom/libtommath/runs/5650310855?check_suite_focus=true

I'll simply leave it out. "It works as it is now"^TM :)

Reply:
> Seems like you fixed it somehow?

yes, by removing this option

Reply:
You're the libtommath pro here, but are `s_read_wincsp` and `s_read_arc4random` exported symbols by the lbrary?
Isn't the issue here that a consumer is using private symbols which the compiler apparently stripped or does not export from the library?

Reply:
Those are static functions which should be dead-code-eliminated and the problem is that clang only does that from `-O1` on.